internal/deviations/deviations.go

```diff
@@ -99,4 +108,21 @@ var (
 	StaticProtocolName = flag.String("deviation_static_protocol_name", "DEFAULT", "The name used for the static routing protocol.  The default name in OpenConfig is \"DEFAULT\" but some devices use other names.")
 
 	DeprecatedVlanID = flag.Bool("deviation_deprecated_vlan_id", false, "Device requires using the deprecated openconfig-vlan:vlan/config/vlan-id or openconfig-vlan:vlan/state/vlan-id leaves.")
+
+	AddSubIntfToNetInst = flag.Bool("deviation_adding_subintf_to_netowork_instance", true,
+	"Vendor requiring explicitly attaching sub-interface to network-instance, this deviation should be marked true. Default expectation is sub-interface is part of default network-instance implicitly.")
 )
+
+func SubIntfToNetworkInstance(t *testing.T, dconf *ondatra.Config, i *telemetry.Interface, si uint32, inst string) {
```

Comment:
Is this the right place to include this code -- pinging @liulk to discuss where we want to include the code that implements deviations, today it's included in the code base, but I think having this as a reusable library would be helpful.

Reply:
This code probably belongs best in https://github.com/openconfig/featureprofiles/blob/main/internal/attrs/attrs.go

Add a function `AddToDUT` that will both replace the interface and put it in a VRF. Add the VRF attribute - if empty, use *deviations.DefaultNetworkInstance.

Reply:
Ack'd - I'll move this as AddToDut in attrs.go - and If the NI is empty, then pass it to default network instance.

Reply:
Please move the functoin to attrs.go. If it's not clear what AddToDUT should do, it should functionally replace these lines in static_arp_test.go:

```
	i1 := &telemetry.Interface{Name: ygot.String(p1.Name())}
	d.Interface(p1.Name()).Replace(t,
		configInterfaceDUT(i1, &dutSrc, &ateSrc, peermac))

	if *deviations.AddSubIntfToNetInst {
		deviations.SubIntfToNetworkInstance(t, d, i1, 0, *deviations.DefaultNetworkInstance)
	}
```

Reply:
@liulk - I'll move it over, I don't know if I'll have it ready before the end of the day today. I'll also need to run a few tests locally. I'll target End of Day Monday for a more realistic timeframe.

Reply:
@liulk, I finally got around to implementing/testing moving the function to AddToDUT. Changes have been pushed to this PR.

Reply:
Resolving Conversation as this was moved to AddToDUT() in `attrs.go`